### ISSUE TITLE
Decoupling: Assemble document sidebar in `wp-story-editor`

### ISSUE THREAD

**Issue by swissspidy** [swissspidy/web-stories-wp#9070]
## Feature Description

As suggested at https://github.com/google/web-stories-wp/pull/9027#issuecomment-922682077:

What if we export all panels from Document and assemble them in `wp-story-editor` like below

```js
import {
  PublishPanel,
  ExcerptPanel,
  SlugPanel,
  StatusPanel,
  PageAdvancementPanel,
  BackgroundAudioPanel,
  TaxonomiesPanel,
} from '@web-stories-wp/story-editor';

const Document = () => {
  function DocumentInspector() {
    const enabledTaxonomies = useFeature('enableTaxonomiesSupport');
    return (
      <>
        <StatusPanel />
        <PublishPanel />
        <ExcerptPanel />
        <SlugPanel />
        <PageAdvancementPanel />
        <BackgroundAudioPanel />
        {enabledTaxonomies ? <TaxonomiesPanel /> : null}
      </>
    );
  }
}

function Layout() {
  return (
    <MetaBoxesProvider>
      <InterfaceSkeleton
        header={<Header />}
        inspector={ {
            document: {
              component: <Document />,
            },
          }
        }
      >
        <MetaBoxes />
      </InterfaceSkeleton>
    </MetaBoxesProvider>
  );
}

```

That will help us in a couple of ways

1. We would have more control over individual components like `PublishPanel`, so we can easily pass more configuration options ( props ) to handle `isSaving` or passing labels, etc.
2. It would be possible to remove/add panels from a third-party integration.

Just a thought that may be later the entire inspector panel may be configured from the third party integration using the following. 

```js
      <InterfaceSkeleton
        header={<Header />}
        inspector={ {
          design: {
            title: __( 'Design', 'web-stories' ),
            component: <Design />
          },
          document: {
            title: __( 'Document', 'web-stories' ),
            component: <Document />,
          },
          ...
        } }
      >
```

## Additional Context

Gutenberg does this in a similar way, see the `SettingsSidebar` component: https://github.com/WordPress/gutenberg/blob/164bef81386e96330128c6218a66f29c9ed8e865/packages/edit-post/src/components/sidebar/settings-sidebar/index.js

**Comment by mohdsayed** [comment]
Yes, but not only this, I think we should move some more pieces of the story editor that do not make sense for a standalone story editor. Remember, when we were working with the [playground](https://rtcamp.github.io/web-stories-playground/#page=%252246ae8de1-2b13-42b8-b60e-31fad9761501%2522) we realized that without a CMS/backend integration, a number of items will not be required for the core-editor ( including most of the items of the documents panel ).

If we look at the Gutenberg block editor, the core functionality of the editor works when you [put the required components together](https://github.com/WordPress/gutenberg/blob/trunk/storybook/stories/playground/index.js). Similarly, the idea is that the basic functionality of core story editor should work without any backend integration.

**Comment by mohdsayed** [comment]
Just for reference, these were the items we removed for the playground

Component | Action | Notes
-- | -- | --
Library: Upload Button | Disable | Since we aren’t giving the option of uploading media to the playground.
Update/Publish/Draft buttons | Remove | In the playground, stories cannot be saved or published.We can safely remove these buttons.For users it should be clear that such options exist in the plugin.
Document Panel: Radio controls of draft/public/private status | Remove | Since stories cannot be saved or published in the playground, adding these options—even when disabled— is confusing.They also take up a lot of space.
Document Panel: Publish date selection | Remove | Irrelevant for playground
Document Panel: Permalink input control | Remove | Irrelevant for playground
Document Panel: Poster Image / Publisher Logo | Remove | Can we set some dummy/default images here just so the resulting story is valid, but prevent changing the images?
Document Panel: Authors Dropdown | Remove | There are no authors to choose from in the playground, so let’s remove this.
Page Templates: Saved Templates option | Remove | Playground is not persistent so this can be removed.
Design Panel : Caption & Subtitles for Video | Disable | Shows when the video layer is selected, not required as the user cannot upload files.
Design Panel: Accessibility : Poster Image Edit Icon | Remove | Allows you to edit poster images. Disable because files cannot be uploaded but the video poster is visible.
Document Panel: Story Description | Remove | Not required for the playground.
Library: Search Field | Remove | Search is based on title, the user would not know the media titles/names and therefore the search field would not be very useful for the playground.
Library: Option to edit media | Remove | Users cannot edit or remove media, so there is no use of this option.

**Comment by swissspidy** [comment]
> Yes, but not only this, I think we should move some more pieces of the story editor that do not make sense for a standalone story editor. 

True, there are definitely more places where we could move logic accordingly.

> Just for reference, these were the items we removed for the playground

Thanks for sharing this table for reference, very useful for discussion 👍 

> Component | Action | Notes
> -- | -- | --
> Library: Upload Button | Disable | Since we aren’t giving the option of uploading media to the playground.

I think we can continue to hide/disable this button/functionality based on user permission checks.

> Component | Action | Notes
> -- | -- | --
> Update/Publish/Draft buttons | Remove | In the playground, stories cannot be saved or published.We can safely remove these buttons.For users it should be clear that such options exist in the plugin.

I think this one is already covered by your refactoring work on the buttons there.

> Component | Action | Notes
> -- | -- | --
> Document Panel: Radio controls of draft/public/private status | Remove | Since stories cannot be saved or published in the playground, adding these options—even when disabled— is confusing.They also take up a lot of space.
> Document Panel: Publish date selection | Remove | Irrelevant for playground
> Document Panel: Permalink input control | Remove | Irrelevant for playground

Covered by this ticket with the goal to assemble the document panel differently.

> Component | Action | Notes
> -- | -- | --
> Document Panel: Poster Image / Publisher Logo | Remove | Can we set some dummy/default images here just so the resulting story is valid, but prevent changing the images?

Also covered by this ticket. We could think about providing default images for platforms other than WP for sure. Preventing changes would fall under media upload checks.

> Component | Action | Notes
> -- | -- | --
> Document Panel: Authors Dropdown | Remove | There are no authors to choose from in the playground, so let’s remove this.

Also covered by this ticket.

> Component | Action | Notes
> -- | -- | --
> Page Templates: Saved Templates option | Remove | Playground is not persistent so this can be removed.

We should probably introduce some capability checks there to cover this. Related: #8665 

> Component | Action | Notes
> -- | -- | --
> Design Panel : Caption & Subtitles for Video | Disable | Shows when the video layer is selected, not required as the user cannot upload files.

We already hide this panel if user doesn't have upload permission and there aren't any captions. So we should be good here.

> Component | Action | Notes
> -- | -- | --
> Design Panel: Accessibility : Poster Image Edit Icon | Remove | Allows you to edit poster images. Disable because files cannot be uploaded but the video poster is visible.

We already hide this option if user doesn't have upload permission

> Component | Action | Notes
> -- | -- | --
> Document Panel: Story Description | Remove | Not required for the playground.

Also covered by this ticket.

> Component | Action | Notes
> -- | -- | --
> Library: Search Field | Remove | Search is based on title, the user would not know the media titles/names and therefore the search field would not be very useful for the playground.

That is a good one! I think we need to revisit whether we really want to hide this search form, or whether we think this is a core feature that is relevant for all platforms (incl. playground). Alternatively this could be some flag passed via config provider.

> Component | Action | Notes
> -- | -- | --
> Library: Option to edit media | Remove | Users cannot edit or remove media, so there is no use of this option.

Also covered by permission checks.

**Comment by swissspidy** [comment]
This was fixed by #9614
